lib/preview-web/src/StoryRender.ts

```diff
@@ -204,10 +209,18 @@ export class StoryRender<TFramework extends AnyFramework> implements Render<TFra
 
       if (forceRemount && playFunction) {
         this.disableKeyListeners = true;
-        await this.runPhase(abortSignal, 'playing', async () =>
-          playFunction(renderContext.storyContext)
-        );
-        await this.runPhase(abortSignal, 'played');
+        try {
+          await this.runPhase(abortSignal, 'playing', async () => {
+            await playFunction(renderContext.storyContext);
+          });
+          await this.runPhase(abortSignal, 'played');
+        } catch (error) {
+          logger.error(error);
+          await this.runPhase(abortSignal, 'errored', async () => {
+            const { name: errorName = 'Error', message = String(error), stack } = error;
+            this.channel.emit(STORY_THREW_EXCEPTION, { name: errorName, message, stack });
```

Comment:
Do you think this should be a separate event?

As it is the same event, does this mean a "true" (non-play) exception is also rendered in the interactions panel (as well as a red box?)

Reply:
You're right, it does, and that's a mistake. In fact if the story throws, the play function shouldn't be run at all I think. Currently it does and it will fail as soon as it attempts to query for a DOM node (which won't exist).

I'll create a separate event and try to avoid running the play function if the story failed to render.